src/app/components/security/pages/register/register.component.ts

```diff
@@ -1,53 +1,82 @@
 import { Component, OnInit } from "@angular/core";
-import { SecurityService } from "@baw-api/security/security.service";
+import { ActivatedRoute, Router } from "@angular/router";
+import {
+  IRegisterDetails,
+  RegisterDetails,
+  SecurityService,
+} from "@baw-api/security/security.service";
 import {
   registerMenuItem,
   securityCategory,
 } from "@components/security/security.menus";
-import { PageComponent } from "@helpers/page/pageComponent";
+import {
+  defaultErrorMsg,
+  FormTemplate,
+} from "@helpers/formTemplate/formTemplate";
 import { ToastrService } from "ngx-toastr";
+import { takeUntil } from "rxjs/operators";
 import { fields } from "./register.schema.json";
 
 @Component({
   selector: "baw-authentication-register",
   template: `
-    <baw-wip>
-      <baw-form
-        title="Register"
-        size="small"
-        [model]="model"
-        [fields]="fields"
-        submitLabel="Register"
-        [submitLoading]="loading"
-        (onSubmit)="submit($event)"
-      ></baw-form>
-    </baw-wip>
+    <baw-form
+      title="Register"
+      size="small"
+      [model]="model"
+      [fields]="fields"
+      submitLabel="Register"
+      [submitLoading]="loading"
+      [seed]="seed"
+      (onSubmit)="submit($event)"
+    ></baw-form>
   `,
 })
-class RegisterComponent extends PageComponent implements OnInit {
-  public model = {};
+class RegisterComponent
+  extends FormTemplate<RegisterDetails>
+  implements OnInit {
   public fields = fields;
   public loading: boolean;
+  public seed: string | boolean = true;
 
   public constructor(
     private api: SecurityService,
-    private notifications: ToastrService
+    notifications: ToastrService,
+    route: ActivatedRoute,
+    router: Router
   ) {
-    super();
+    super(
+      notifications,
+      route,
+      router,
+      undefined,
+      () => "Successfully registered new account",
+      defaultErrorMsg,
+      false
+    );
   }
 
   public ngOnInit() {
+    super.ngOnInit();
+
     if (this.api.isLoggedIn()) {
       // Disable submit button
       this.loading = true;
       this.notifications.error("You are already logged in.");
+      return;
     }
+
+    this.api
+      .signUpSeed()
```

Comment:
we'll probably want to standardize this call to `new()` once we remove the hacks (we'll still need recapcha seeds even with functional REST APIs). Not important for now though.